eis_toolkit/exploratory_analyses/parallel_coordinates.py

```diff
@@ -0,0 +1,192 @@
+from typing import Optional
+
+import matplotlib
+import matplotlib.colors as mcolors
+import matplotlib.patches as patches
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from beartype import beartype
+from beartype.typing import Tuple
+from matplotlib.cm import ScalarMappable
+from matplotlib.path import Path
+from sklearn.preprocessing import LabelEncoder
+
+from eis_toolkit import exceptions
+
+
+def _normalize_data(data) -> Tuple[np.ndarray, float, float]:
+    y_min = data.min(axis=0)
+    y_max = data.max(axis=0)
+    dy = y_max - y_min
+    y_min -= dy * 0.05
+    y_max += dy * 0.05
+    dy = y_max - y_min
+
+    normalized_data = np.zeros_like(data)
+    normalized_data[:, 0] = data[:, 0]
+    normalized_data[:, 1:] = (data[:, 1:] - y_min[1:]) / dy[1:] * dy[0] + y_min[0]
+
+    return normalized_data, y_min, y_max
+
+
+def _get_default_palette(color_data_numeric):
+    if color_data_numeric:
+        return "Spectral"
+    else:
+        return "bright"
+
+
+@beartype
+def _plot_parallel_coordinates(
+    data: np.ndarray,
+    data_labels: np.ndarray,
+    color_data: np.ndarray,
+    color_column_name: str,
+    plot_title: Optional[str],
+    palette_name: Optional[str],
+    curved_lines: bool,
+) -> matplotlib.figure.Figure:
+
+    fig, main_axis = plt.subplots()
+
+    # If color_data is not numeric, encode the color data to create colors from palette correctly
+    if not np.issubdtype(type(color_data[0]), np.number):
+        color_data_numeric = False
+        label_encoder = LabelEncoder()
+        color_data = list(label_encoder.fit_transform(color_data))
+    else:
+        label_encoder = None
+        color_data_numeric = True
+
+    # If palette name is not provided, go with the default
+    palette_name = _get_default_palette(color_data_numeric) if not palette_name else palette_name
+
+    # Normalize data for drawing lines
+    normalized_data, y_min, y_max = _normalize_data(data)
+
+    # Set style
+    axes_list = [main_axis] + [main_axis.twinx() for _ in range(normalized_data.shape[1] - 1)]
+    for i, axis in enumerate(axes_list):
+        axis.set_ylim(y_min[i], y_max[i])
+        axis.spines["top"].set_visible(False)
+        axis.spines["bottom"].set_visible(False)
+        if axis != main_axis:
+            axis.spines["right"].set_visible(False)
+            axis.yaxis.set_ticks_position("left")
+            axis.spines["left"].set_position(("axes", i / (normalized_data.shape[1] - 1)))
+
+    main_axis.set_xlim(0, normalized_data.shape[1] - 1)
+    main_axis.set_xticks(range(normalized_data.shape[1]))
+    main_axis.set_xticklabels(data_labels, fontsize=14)
```

Comment:
Longer labels overlap (see penguins plot in the notebook). How about rotating and decreasing fontsize?
```suggestion
    main_axis.set_xticklabels(data_labels, fontsize=10, rotation=45)
```
![Screenshot from 2023-07-25 07-58-49](https://github.com/GispoCoding/eis_toolkit/assets/134925837/b7f222b1-6843-4ef2-a245-7bbde901e589)

Reply:
Or increase figure size (`fig, main_axis = plt.subplots(figsize=(15, 15)` at row 52, for example)

Reply:
I only changed to fontsize to 10 for now, that seemed to give already quite a lot more space for the feature labels.